File: mentees/hareshkat/algorithms/problem1.py
"""
Problem 1 :-
You are climbing a staircase. It takes n steps to reach the top.
Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?
"""

def get_step_count(n: int):
    """
    Recursion function to get the fibonacci numbers
    :param n: Integer number
    :return: Sum of last two calls

    """
    if n <= 1:
        return n
    return get_step_count(n - 1) + get_step_count(n - 2)


if __name__ == "__main__":
    n = 20 #Given Number
    step_counts = get_step_count(n)
    print(f"Number of ways to climb {n} stpes is {step_counts}")

File: mentees/bhavinchauhan98/tests/problem_1.js
// Problem 1 :-

// You are climbing a staircase. It takes n steps to reach the top.

// Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?

// Example 1:

// Input: n = 2
// Output: 2
// Explanation: There are two ways to climb to the top.
// 1. 1 step + 1 step
// 2. 2 steps
// Example 2:

// Input: n = 3
// Output: 3
// Explanation: There are three ways to climb to the top.
// 1. 1 step + 1 step + 1 step
// 2. 1 step + 2 steps
// 3. 2 steps + 1 step

function climbingSteps(n) {
    if (n >= 0) {
        if (n === 0) return 1
        return climbingSteps(n-1) + climbingSteps(n-2)
    }
    return 0
}

function main() {
    const n = 3

    return climbingSteps(n)
}

console.log(main())